parser_test.go

```diff
@@ -4585,6 +4585,22 @@ func (s *testParserSuite) TestFulltextSearch(c *C) {
 	c.Assert(writer.String(), Equals, "MATCH(title,content) AGAINST(\"search\" WITH QUERY EXPANSION)")
 }
 
+func (s *testParserSuite) TestStartTransaction(c *C) {
+	cases := []testCase{
+		{"START TRANSACTION READ WRITE", true, "START TRANSACTION"},
+		{"START TRANSACTION READ ONLY", true, "START TRANSACTION READ ONLY"},
+		{"START TRANSACTION READ ONLY WITH TIMESTAMP BOUND", false, ""},
+		{"START TRANSACTION READ ONLY WITH TIMESTAMP BOUND STRONG", true, "START TRANSACTION READ ONLY WITH TIMESTAMP BOUND STRONG"},
+		{"START TRANSACTION READ ONLY WITH TIMESTAMP BOUND MAX STALENESS '00:00:10'", true, "START TRANSACTION READ ONLY WITH TIMESTAMP BOUND MAX STALENESS '00:00:10'"},
```

Comment:
Is this SQL a valid one?

```
START TRANSACTION READ ONLY WITH TIMESTAMP BOUND MAX STALENESS 1 + 1
```

Reply:
`STALENESS` should take an expression that can be evaluated to `TIME`.
`TIMESTAMP` would take an expression that can be evaluated to `DATETIME` or `TIMESTAMP` on the other hand.